Calvin Passmore

A02107892

ECE 5420

# Exam: Analog Circuits

---


## Problem 1

Triode Mode: id = Kn[Vov * Vds - 0.5 * Vds^2] * (1 + &lambda; * Vds)

id = 1.5 mA = (W/L) * Ut * [Vov * Vds - 0.5 * Vds ^ 2] * (1 + &lambda; + Vds)

i_target = id * 10

Kn_target * Constants = Kn * Constants * 10

(W / L) = (W / L ) * 10

w_target = w * 10

w = 10 um

The simulation matched the calculations fairly closely but was not exact, the simulation gave a width of 11.3 um.

The calculated results:
![1calc](./1_calc.png)

Adjusting the simulation to give a perfect 15mA:
![1sim](./1_sim.png)

---
---


## Problem 2

The NMOS is now in saturation because Vds changed. Vov = Vgs - Vth = 3 - 0.43 = 2.57. Vds = Vdd - Vled = 9 - 2 = 7. Vds > Vov &rightarrow; Saturation.

In [12]:
W = 1e-6
L = 0.2e-6
Vgs = 3
Vth = 0.43
Vov = Vgs - Vth
Ut = 118e-6
lam = 0.06
Vds = 7.21  # from EveryCircuit - LED voltage drop

id = 0.5 * W / L * Ut * (Vov**2) * (1 + (lam * Vds))
print(id)

0.0027913430232999995


In [15]:
i_target = 15e-3
w_target = W * (i_target / id)
print(w_target)

5.373757318535004e-06



Using the ratio of the calculated current, and the desired current, we can get a target width of 5.37 um. Verifying with EveryCircuit gives a width of 5.343

![2](./2.png)

If Vds is changed to 12, and 24. The currents are as follows

In [19]:
V12 = 12 - 2 # subtract 2 for the LED
V24 = 24 - 2

i12 = 0.5 * w_target / L * Ut * (Vov**2) * (1 + (lam * V12))
i24 = 0.5 * w_target / L * Ut * (Vov**2) * (1 + (lam * V24))

print(i12)
print(i24)

0.01675275722462655
0.024291497975708495


To explain my prediction, we just changed the Vds value in accordance with the channel length modulation.

Vds = 12v &rightarrow; id = 16.75 mA
Vds = 24v &rightarrow; id = 24.29 mA

The EveryCircuit demo gives value of

Vds = 12v &rightarrow; id = 16.9 mA
Vds = 24v &rightarrow; id = 24.5 mA

The small discrepency can be attributed to the slight difference of the new w value

---
---

## Problem 3

Maximum signal amplitude:

In [21]:

Vin_max = 2 # v
DC = 2  # v
iph_max = 100e-6 * (Vin_max + DC)

print(iph_max)

0.0004


Max signal amplitude = 400 uA

EveryCircuit max signal amplitude = 400 uA

Min Voltage = Vdd - Imax*R = 5 - 400u * 10k = 5- 4 = 1 v

![](./3_dc_offset.png)

Now adding the ambient light as 2V DC

In [22]:
DC = 4
iph_max = 100e-6 * (Vin_max + DC)

print(iph_max)

0.0006000000000000001


Max signal amplitude = 600 uA

Everycircuit max signal amplitude = 563 uA

Min Voltage = Vdd - (Imax * R - (Iled)) = 5 - (600u * 10k - Iled) = 5 - 6 - Iled * R = -1 * Iled*R v. So this would be -1 v except it activates the forward bias of the photodiode.

![](./3_ambient.png)

The difference is that when the current gets large enough, its creating a negative voltage (creating more than 5V) so the photodiode is becoming forward biased and letting current through that offsets some of the light driven current.

Generalized equation:

Imax = 100uA * (Vin_max + DC) when Vin_max is small

Imax = 100uA * (Vin_max + DC) - Iled when Vin_max is close to or larger than VDD

### Attach a load

When a load is attached, it consumes some of the current at the Vout, decreasing Vout. This prevents Vout from getting close to the LED forward bias voltage and not allowing the photodiode to  have voltage

![](./3_load.png)

## Problem 4

Advantage:
+ Can maintain a higher amplitude without a load, decouples the output from VDD

Disadvantage:
+ More parts

Maximum Signal Amplitude

![](./maxamp_4.drawio.png)

The above drawing shows that when the current is moved through the resistor the voltage at the end of the op amp will be fluctuating at Iph * R + 1

Imax = 100u * (Vmax + DC) = 100u * (2 + 2) = 400 u

![](./4_dc.png)

Imax = 100u * (Vmax + DC) = 100u * (2 + 4) = 600u

Generalized Imax = Gain * (Vmax + DC_ambient)

### Attach a load

When a load is attached, the current passing through the photodiode, and the resistor, is not changed. Therefore Vout is not changed when the load is applied. This is a huge advantage, that the load is decoupled from Vout.

![](./4_load.png)

### Differences

The biggest differences, besides the components, is that the load has been decoupled from Vout. So Vout is no longer affected by what is happening at the load.

## Problem 5

![](./5_vadj0.drawio.png)

Ib = 0.5 * Kp * Vov^2 = 0.5 * Kp * (Vgs - Vthp) ^ 2 = 0.5 * Kp * (Vx - Vthp) ^ 2

Vx = R * (Ib + Ia)

0 = Vx^2 * Kp - Vx / R - Kp * Vthp + Kp * Vthp^2 + Ia

In [34]:
from math import sqrt

Kp = 30.3 # From EveryCircuit
R = 1e3
Vthp = 400e-3 # From EveryCircuit

Vin_max = 2
Vdc = 2

Ia = 100e-6 * (Vin_max + Vdc)

A = Kp
B = 1 / R
C = -Kp * Vthp + Kp * Vthp**2 + Ia

Vx1 = (-B + sqrt(B**2 - 4 * A * C)) / (2 * A)
Vx2 = (-B - sqrt(B**2 - 4 * A * C)) / (2 * A)

print(Vx1)
print(Vx2)

0.48986797345817357
-0.48990097675850364


Vx = 0.490 v

In [36]:
Ib = 0.5 * Kp * ((Vx1 - Vthp)**2)
print(Ib)

0.12235522770020663


IB = 122.4 mA

### Generalize Vout

Vout = (Iph * R) + Vadj = (100u * (Vin_max + V_dc_ambient) * R) + Vadj


### What is the PMOS doing

The PMOS is keeping the photodiode in reverse bias. This allows Vadj to be changed to the necessary value while keeping the photodiode in reverse bias.

### Loaded Vout

When a load is applied to Vout, Vout is unchanged. This is done by keeping the photodiode in reverse bias, as well as isolating Vout from the rest of the circuit with the OpAmp.

![](./5Vadj_volt_current.png)

## Problem 6

This circuit works by charging/discharging the capacitor to keep the output of the OpAmp at the appropriate 'Vadj' level, offsetting the output voltage to keep Vout centered at 0v.

Using the small signal analysis near Vout, it gives us this.

![](./6_small_sig.drawio.png)

Using current divisors, Iph = 10k/11k * Iop and Iadj = 1k/11k * Iop

Iop = 11/10 * Iph = 11/10 * (Iph * (Vinmax + Vdc)) = 11/10 * 100u * (4) = 440uA

Iadj = 1/10 * 100u * (4) = 40uA

Iph = 100u * (4) = 400uA

Vout = Iadj * Radj = 40uA * 1k = 

![](./6_amp_change.png)
